redis/pool.go

```diff
@@ -270,8 +280,17 @@ func (p *Pool) get(ctx interface {
 }) (*poolConn, error) {
 
 	// Handle limit for p.Wait == true.
+	var waited time.Duration
 	if p.Wait && p.MaxActive > 0 {
 		p.lazyInit()
+
+		// wait indicates if we believe it will block so its not 100% accurate
+		// however for stats it should be good enough.
+		wait := len(p.ch) == 0
```

Comment:
Inherently racey but if you want to just get wait stats when you think the channel will block, not sure there is any other way.

Is there any reason you can't do the wait stat calculation all the time? It'll be ~0 in non-blocking cases so maybe not much use?

Reply:
It would just break the count as every request would flagged as waited.

Reply:
Yeh thats not ideal